tasks/shared_libraries/recovery/shared_recovery.py

```diff
@@ -5,86 +5,106 @@
+    db_queue_url: str,
+):
+
+    new_data = {"job_id": job_id, "granule_id": granule_id}
     if request_time is not None:
-        new_data['request_time'] = request_time
+        new_data["request_time"] = request_time
     if completion_time is not None:
-        new_data['completion_time'] = completion_time
+        new_data["completion_time"] = completion_time
     if archive_destination is not None:
-        new_data['archive_destination'] = archive_destination
-
-    post_entry_to_queue('orca_recoveryjob',
-                        new_data,
-                        request_method, db_queue_url)
-
-def post_status_for_file_to_queue(job_id: str, granule_id: str, filename: str, key_path: Optional[str],
-                                  restore_destination: Optional[str],
-                                  status_id: OrcaStatus, error_message: Optional[str],
-                                  request_time: Optional[str], last_update: str,
-                                  completion_time: Optional[str],
-                                  request_method: RequestMethod,
-                                  db_queue_url: str):
-    new_data = {'job_id': job_id,
-                'granule_id': granule_id,
-                'filename': filename}
+        new_data["archive_destination"] = archive_destination
+
+    post_entry_to_queue("orca_recoveryjob", new_data, request_method, db_queue_url)
+
+
+def post_status_for_file_to_queue(
+    job_id: str,
+    granule_id: str,
+    filename: str,
+    key_path: Optional[str],
+    restore_destination: Optional[str],
+    status_id: OrcaStatus,
+    error_message: Optional[str],
+    request_time: Optional[str],
+    last_update: str,
+    completion_time: Optional[str],
+    request_method: RequestMethod,
+    db_queue_url: str,
+):
+    new_data = {"job_id": job_id, "granule_id": granule_id, "filename": filename}
     if key_path is not None:
-        new_data['key_path'] = key_path
+        new_data["key_path"] = key_path
     if restore_destination is not None:
-        new_data['restore_destination'] = restore_destination
+        new_data["restore_destination"] = restore_destination
     if error_message is not None:
-        new_data['error_message'] = error_message
+        new_data["error_message"] = error_message
     if request_time is not None:
-        new_data['request_time'] = request_time
+        new_data["request_time"] = request_time
     if last_update is not None:
-        new_data['last_update'] = last_update
+        new_data["last_update"] = last_update
     if completion_time is not None:
-        new_data['completion_time'] = completion_time
+        new_data["completion_time"] = completion_time
+
+    post_entry_to_queue("orca_recoverfile", new_data, request_method, db_queue_url)
+
+
+def post_entry_to_queue(
+    table_name: str,
+    new_data: Dict[str, Any],
+    request_method: RequestMethod,
+    db_queue_url: str,
+):
+    body = json.dumps(new_data)
```

Comment:
Add doc string. See previous comments on other items to cleanup in the shared library.

Review with @AndrewDorn  if tablename variable still needed.

Reply:
@AndrewDorn to provide the doc string